File: 02-programacion-basica/02-variables-constantes-y-tipos/java/soluciones/RetoCompraIgv.java
public class RetoCompraIgv {
    public static void main(String[] args) {
        double precio = 25.0;
        int cantidad = 3;
        final double IGV = 0.18;

        double subtotal = precio * cantidad;
        // La constante expresa una regla que no cambia durante el cálculo.
        double impuesto = subtotal * IGV;
        double total = subtotal + impuesto;

        System.out.printf("Subtotal: %.2f | IGV: %.2f | Total: %.2f%n", subtotal, impuesto, total);
    }
}

File: 02-programacion-basica/02-variables-constantes-y-tipos/scala/soluciones/RetoCompraIgv.scala
@main def retoCompraIgv(): Unit =
  // val mantiene referencias inmutables durante el cálculo.
  val precio = 25.0
  val cantidad = 3
  val igv = 0.18

  val subtotal = precio * cantidad
  val impuesto = subtotal * igv
  val total = subtotal + impuesto
  println(f"Subtotal: $subtotal%.2f | IGV: $impuesto%.2f | Total: $total%.2f")
